chainercv/links/model/feature_extraction_predictor.py

```diff
@@ -0,0 +1,161 @@
+import numpy as np
+import warnings
+
+import chainer
+from chainer import cuda
+
+from chainercv.transforms import center_crop
+from chainercv.transforms import resize
+from chainercv.transforms import scale
+from chainercv.transforms import ten_crop
+
+
+class FeatureExtractionPredictor(chainer.Chain):
+
+    """Wrapper that adds a prediction method to a feature extraction model.
+
+    The :meth:`predict` takes three steps to make a prediction.
+
+    1. Preprocess input images
+    2. Forward the preprocessed images to the network
+    3. Average features in the case when more than one crops are extracted.
+
+    Example:
+
+        >>> from chainercv.links import VGG16
+        >>> from chainercv.links import FeatureExtractionPredictor
+        >>> base_model = VGG16()
+        >>> model = FeatureExtractionPredictor(base_model)
+        >>> prob = model.predict([img])
+        # Predicting multiple features
+        >>> model.extractor.feature_names = ['conv5_3', 'fc7']
+        >>> conv5_3, fc7 = model.predict([img])
+
+    When :obj:`self.crop == 'center'`, :meth:`predict` extracts features from
+    the center crop of the input images.
+    When :obj:`self.crop == '10'`, :meth:`predict` extracts features from
+    patches that are ten-cropped from the input images.
+
+    When extracting more than one crops from an image, the output of
+    :meth:`predict` returns the average of the features computed from the
+    crops.
+
+    Args:
+        extractor: A feature extraction model. This is a callable chain
+            that takes a batch of images and returns a variable or a
+            tuple of variables.
```

Comment:
How about adding an argument `mean`?
If `mean` is None, `FeatureExtractorPredictor` tries to use extractor's attribute.
If `mean` is not None, it uses the argument.